cpp/src/parquet/arrow/schema.cc

```diff
@@ -1017,7 +1016,10 @@ Result<bool> ApplyOriginalMetadata(const Field& origin_field, SchemaField* infer
 
       // Restore extension type, if the storage type is the same as inferred
       // from the Parquet type
-      if (ex_type.storage_type()->Equals(*inferred->field->type())) {
+      if (ex_type.storage_type()->Equals(*inferred->field->type()) ||
+          ((ex_type.extension_name() == "arrow.json") &&
+           ::arrow::extension::JsonExtensionType::IsSupportedStorageType(
```

Comment:
The `IsSupportedStorageType` condition should always be true here, since `ex_type` is a JSONExtensionType and `inferred->field->type` is equal to `ex_type`'s storage type.

Reply:
Huh, yes `IsSupportedStorageType` is indeed redundant here. Removed.